invokeai/backend/stable_diffusion/extensions/inpaint.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+import einops
+import torch
+from diffusers import UNet2DConditionModel
+
+from invokeai.backend.stable_diffusion.extension_callback_type import ExtensionCallbackType
+from invokeai.backend.stable_diffusion.extensions.base import ExtensionBase, callback
+
+if TYPE_CHECKING:
+    from invokeai.backend.stable_diffusion.denoise_context import DenoiseContext
+
+
+class InpaintExt(ExtensionBase):
+    def __init__(
+        self,
+        mask: torch.Tensor,
+        is_gradient_mask: bool,
+    ):
```

Comment:
Add a docstring explaining these params. Here's a draft:
```suggestion
    def __init__(
        self,
        mask: torch.Tensor,
        is_gradient_mask: bool,
    ):
        """Initialize InpaintExt.
        Args:
            mask (torch.Tensor): The inpainting mask. Shape: (batch_size, 1, latent_height, latent_width). Values are
                expected to be in the range [0, 1]. A value of 0 means that the corresponding 'pixel' should not be
                inpainted.
            is_gradient_mask (bool): If True, mask is interpreted as a gradient mask meaning that the mask values range
                from 0 to 1. If False, mask is interpreted as binary mask meaning that the mask values are either 0 or
                1.
        """
```

I wrote this based on what I _thought_ these fields represented but based on testing this is wrong. The mask contains values in the range [0, 2]. Is this a bug in the code that prepares the mask? Or did I misunderstand the intended mask representation? We should get to the bottom of this.

Reply:
It's because gradient mask outputs mask in range [-1,1] I'll include fix of this in this PR.